lib/matplotlib/collections.py

```diff
@@ -899,6 +900,118 @@ def set_paths(self, paths):
     def get_paths(self):
         return self._paths
 
+    def legend_items(self, mode="colors", useall="auto", num=10,
+                     fmt=None, func=lambda x: x, **kwargs):
+        """
+        Creates legend handles and labels for a PathCollection. This is useful
+        for obtaining a legend for a :meth:`~.Axes.scatter` plot. E.g.::
+
+            scatter = plt.scatter([1,2,3], [4,5,6], c=[7,2,3])
+            plt.legend(*scatter.legend_items())
+
+        Also see the :ref:`automatedlegendcreation` example.
+
+        Parameters
+        ----------
+        mode : string, optional, default *"colors"*
```

Comment:
Not sure I like "mode" here, though I guess its OK.  "key" would be more natural to me...

Reply:
What makes `key`more natural? Any good argument for `key`?  
In principle I'm open to other names here, e.g. `prop` because colors or sizes are properties?

Reply:
Yeah, `prop` works for me.